bots-deployment-detector/src/agent.spec.ts

```diff
@@ -0,0 +1,159 @@
+    describe('Ignores Transaction', () => {
+      it('ignores any transaction that is not a from deployer Address', async () => {
+        const txEvent = new TestTransactionEvent();
+        const findings = await handleTransaction(txEvent);
+
+        expect(findings).toStrictEqual([]);
+      });
+
+      it('ignores a transaction that is from a deployer Address but not a bot deployment', async () => {
+        const txEvent = new TestTransactionEvent();
+        txEvent.setFrom(mockDeployerAdddress).setTo(mockRandomAddress).setValue('100000');
+
+        const findings = await handleTransaction(txEvent);
+
+        expect(findings).toStrictEqual([]);
+      });
+
+      it('ignores a transaction that is from a deployer Address and bot deployment but to a different bot registry/contract', async () => {
+        const txEvent = new TestTransactionEvent();
+        txEvent.setFrom(mockDeployerAdddress).setTo(mockRandomAddress);
+        txEvent.addEventLog(eventAgentUpdated, mockRandomAddress, [
+          mockAgentId,
+          mockDeployerAdddress,
+          '',
+          [mockChainId],
+        ]);
+
+        const findings = await handleTransaction(txEvent);
+
+        expect(findings).toStrictEqual([]);
+      });
+    });
+
+    describe('Returns a Finding', () => {
+      let txEvent: TestTransactionEvent;
+
+      const updateInputs = [mockAgentId, mockDeployerAdddress, '', [mockChainId]];
+      const enableInputs = [mockAgentId, true, 1, true];
+      const disableInputs = [mockAgentId, false, 1, false];
+
+      const findingInput = {
+        name: 'Bot Deployment Detector',
+        description: 'Bot deployment/upgrade detected',
+        alertId: 'FORTA-1',
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+      };
+
+      beforeEach(() => {
+        txEvent = new TestTransactionEvent();
+        txEvent.setFrom(mockDeployerAdddress).setTo(mockContractAddress);
+      });
+
+      it('returns a finding with metadata.type: "AgentUpdated" when the transaction is from deployer to update/upgrade of a bot', async () => {
+        txEvent.addEventLog(eventAgentUpdated, mockContractAddress, updateInputs);
+
+        const findings = await handleTransaction(txEvent);
+
+        expect(findings).toStrictEqual([
+          Finding.fromObject({
+            ...findingInput,
+            metadata: {
+              agentId: mockAgentId,
+              type: eventTypes[1],
+              by: mockDeployerAdddress,
+              chainIds: mockChainId.toString(),
+            },
+          }),
+        ]);
+      });
+
+      it('returns a finding with metadata.type: "AgentCreated" when transaction is from deployer for the deployment of the bot', async () => {
+        txEvent.addEventLog(eventAgentUpdated, mockContractAddress, updateInputs);
+
+        const methodInterface = new ethers.utils.Interface(createAgentABI);
+
+        const Tdata = methodInterface.encodeFunctionData('createAgent', [
```

Comment:
similar to the comment above, you can use `addTraces` method of `TestTransactionEvent` which will make this easier.